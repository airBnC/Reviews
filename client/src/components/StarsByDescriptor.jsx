import React from 'react';
import PropTypes from 'prop-types';
import StarRatings from 'react-star-ratings';

const StarsByDescriptor = props => (
  <StarRatings
    rating={props.avg}
    starDimension="18px"
    starSpacing="2px"
    starRatedColor="#008489"
  />
);

StarsByDescriptor.propTypes = { avg: PropTypes.number };

StarsByDescriptor.defaultProps = { avg: 1 };

export default StarsByDescriptor;
