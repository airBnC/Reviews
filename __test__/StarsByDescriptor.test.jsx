import React from 'react';
import { configure, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import StarsByDescriptor from '../client/src/components/StarsByDescriptor';

configure({ adapter: new Adapter() });

const dummyAvg = 4.4;

describe('StarsByDescriptor', () => {
  it('should render correctly', () => {
    const output = shallow(<StarsByDescriptor avg={dummyAvg}/>);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
