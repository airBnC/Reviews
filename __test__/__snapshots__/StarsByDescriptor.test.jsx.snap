// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarsByDescriptor should render correctly 1`] = `
<StarRatings
  changeRating={null}
  gradientPathName=""
  ignoreInlineStyles={false}
  numberOfStars={5}
  rating={4.4}
  starDimension="18px"
  starEmptyColor="rgb(203, 211, 227)"
  starHoverColor="rgb(230, 67, 47)"
  starRatedColor="#008489"
  starSpacing="2px"
  svgIconPath="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
  svgIconViewBox="0 0 51 48"
  typeOfWidget="Star"
/>
`;
